Automattic/sensei
# Apply pattern to the post when selected

Fixes #5140
Fixes #5130
Fixes #5133

### Changes proposed in this Pull Request

* This PR has some general fixes (sorry for that in this PR), but the main goal is to apply the pattern correctly when it's chosen.
* It also finalizes the pattern steps for course and lesson with the proper styles and content.

### Testing instructions


* Create a new course.
* In the wizard form, add a title and a description.
* Navigate through the steps until the "Course Layout".
* Click in a pattern.
* Make sure it's applied to the course with the title and description replaced, and the course is saved.
* Repeat the test, but close the modal instead of choosing a pattern.
* Make sure the modal is closed, the default template is used in the course, and the course is saved.
* Also make sure that the pattern selection work on a new lesson.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/170561649-2ce9480c-d3b8-4de7-aa90-2d9696f6a4ac.mov
